compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -2972,30 +2991,21 @@ fn eval_bin_op_with_integer_literals(
 ) -> Result<Value, Error> {
     fn eval_integer_div(lhs_int: i64, rhs_int: i64, span: PackageSpan) -> Result<Value, Error> {
         match (lhs_int, rhs_int) {
-            (_, 0) => Err(Error::EvaluationFailed(
-                "division by zero".to_string(),
-                span,
-            )),
+            (_, 0) => Err(EvalError::DivZero(span).into()),
             (lhs, rhs) => Ok(Value::Int(lhs / rhs)),
         }
     }
 
     fn eval_integer_mod(lhs_int: i64, rhs_int: i64, span: PackageSpan) -> Result<Value, Error> {
         match (lhs_int, rhs_int) {
-            (_, 0) => Err(Error::EvaluationFailed(
-                "division by zero".to_string(),
-                span,
-            )),
+            (_, 0) => Err(EvalError::DivZero(span).into()),
             (lhs, rhs) => Ok(Value::Int(lhs % rhs)),
         }
     }
 
     fn eval_integer_exp(lhs_int: i64, rhs_int: i64, span: PackageSpan) -> Result<Value, Error> {
         let Ok(rhs_int_as_u32) = u32::try_from(rhs_int) else {
-            return Err(Error::EvaluationFailed(
```

Comment:
is this just a drive-by update?

Reply:
Ugh... the pull request extension ate all my comment responses again...

Yes, this is just some unrelated clean up. We had added the ability to convert directly from normal evaluator errors into partial eval errors late in the process, but then missed a few places that still used hard-coded strings rather then the matching evaluator errors. This cleans up the last of these.